query/aggregate/aggregate.go

```diff
@@ -72,17 +74,8 @@ func groupBy(list api.SeriesList, tags []string) []group {
 	return result
 }
 
-type aggregate int
-
-const (
-	sumAggregate aggregate = iota + 1
-	meanAggregate
-	minAggregate
-	maxAggregate
-)
-
-var aggregateMap = map[aggregate]func([]float64) float64{
-	sumAggregate:
+var aggregateMap = map[string]func([]float64) float64{
```

Comment:
we can make this code more "modular" by instead of hard-coding `map[name]function`, expose a function called `NewAggregate(name string, func function)` which can be used to add functions. In that case, you can define these aggregate functions in different party, and you can compose them that way.